### PR TITLE
Fix misuse of hostname/address in gpinitsystem & gpinitstandby

### PR DESCRIPTION
**Pull request by gfphoenix78**
`hostname` and `address` are different in gp_segment_configuration.
They should be correctly initialized in gpinitsystem. gpinitstandby
is used in gpinitsystem if a standby coordinator is requrested.
gpinitstandby accepts one option to evaluate hostname/address.
The value is interpreted as `gp_segment_configuration.address`,
while the `gp_segment_configuration.hostname` is the hostname
of the deployed node of the standby coordinator.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by gfphoenix78** [comment]
This commit is to fix gpinitsystem and gpinitstandby in the initialization. It only fixes the input configuration for gpinitsystem.
Another commit(https://github.com/gfphoenix78/gpdb/commit/83e9b45fdf4a0aa72739c23cf358b8d680c35ab4) is used to create an isolated network environment for all segments, so we can examine the hostname and address in gp_segment_configuration. It may be included in this PR for test in pipeline.

File: gpMgmt/bin/gpinitstandby [hunk 1]
```diff
@@ -413,7 +430,7 @@ def create_standby(options):
         logger.info('Updating pg_hba.conf file...')
         update_pg_hba_conf(options, array)
         logger.debug('Updating pg_hba.conf file on segments...')
-        update_pg_hba_conf_on_segments(array, options.standby_host, options.hba_hostnames, unreachable_hosts)
+        update_pg_hba_conf_on_segments(array, options.standby_hostname, options.hba_hostnames, unreachable_hosts)
```
Comment (shrakesh) [inline gpMgmt/bin/gpinitstandby:433]: This changes may conflict with PR https://github.com/greenplum-db/gpdb/pull/12806/
which is related to  improvement of pg_hba update for standby.

File: gpMgmt/bin/gpinitstandby [hunk 2]
```diff
@@ -113,8 +113,8 @@ def parseargs():
 
     # Standby initialization options section
     optgrp = OptionGroup(parser, 'Standby initialization options')
-    optgrp.add_option('-s', '--standby-host', type='string', dest='standby_host',
-                      help='hostname of system to create standby coordinator on')
+    optgrp.add_option('-s', '--standby-address', '--standby-host', type='string', dest='standby_address',
```
Comment (piyushc01) [inline gpMgmt/bin/gpinitstandby:116]: Can we add a behave/unit test to make sure 

- '--standby-address' option initialises cluster properly 
- expected entries in gp_segment_confguration are correctly populated.

**Comment by piyushc01** [comment]
Can you please confirm that concourse flow for gpinitstandby and gpinitsystem are green and all tests are passing.

**Comment by wenwangBJ** [comment]
After review the PR, doc and discussion carefully. I don’t think hostname and address of gp_segment_configuration should have any extra meaning related with external  or internal connection. they just mean host or segment. Users may change it if they have multiple NICs for different purpose. If so, they should take any risk by themselves. Greenplum don’t need to cover it.

The only key fix that we need is to use address instead of hostname for segment connections (interconnection motion, ETL and WALreplication).

Based on that, this PR has changed much more than needed. The root casue is quite clear as mentiond by Hongxu Ma  : “we should use SegmentAddress (not SegmentHostName) for replication”

Because it has a lot of utilities that rely on this, my suggestion is to close this existing PR and re-assign this bug with another PR that have more experienced with utility background people.

**Pull request closed without merging**